### PR TITLE
Edit Quest: Cascade Badge + Ticket SS.Anne

### PR DESCRIPTION
**Pull request by Rympex**

### CODE REVIEW

**Comment by Rympex** [comment]
Tested :) and it work without problem, leveling --> get Badge --> get nugget and other items --> get ticket from bill --> and end the quest on route 5 :)

File: Quests/Kanto/CascadeBadgeQuest.lua [hunk 1]
```diff
@@ -8,13 +8,17 @@ local game   = require "Libs/gamelib"
...
-local description = 'Get Cerulean City Badge'
-local level       = 20
+local name        = 'Cascade Badge - Cerulean City: '
+local description = 'Get Badge + Ticket SS.Anne'
+local level       = 21
 
 local dialogs = {
-	fossileGuyBeaten = Dialog:new({
-		"Did you get the one you like?"--,
+	BillTicketDone = Dialog:new({
```
Comment (g0ldPRO) [inline Quests/Kanto/CascadeBadgeQuest.lua:16]: BillTicketDone is an instance of an object, it starts with a lower case character.

File: Quests/Kanto/CascadeBadgeQuest.lua [hunk 2]
```diff
@@ -8,13 +8,17 @@ local game   = require "Libs/gamelib"
...
+		"Have you enjoyed the cruise"--,
+		--""
```
Comment (g0ldPRO) [inline Quests/Kanto/CascadeBadgeQuest.lua:18]: You can remove the comments.

File: Quests/Kanto/CascadeBadgeQuest.lua [hunk 3]
```diff
@@ -26,28 +30,39 @@ function CascadeBadgeQuest:new()
 end
 
 function CascadeBadgeQuest:isDoable()
-	if not hasItem("Cascade Badge") and self:hasMap()
-	then
+	if self:hasMap() then
```
Comment (g0ldPRO) [inline Quests/Kanto/CascadeBadgeQuest.lua:33]: `hasItem("Cascade Badge")` is relevant there. The player will have to come back to this map later, the Cascade Badge is the only information that will allow us to select the right map.
The next Quest should start from the Cerulean Arena, even so this quest stops at Route 5.

File: Quests/Kanto/CascadeBadgeQuest.lua [hunk 4]
```diff
@@ -26,28 +30,39 @@ function CascadeBadgeQuest:new()
...
-	if hasItem("Cascade Badge") then
+	if getMapName() == "Route 5" then
 		return true
 	else
 		return false
 	end
 end
 
-function CascadeBadgeQuest:CeruleanCity()
+function CascadeBadgeQuest:CeruleanCity()
```
Comment (g0ldPRO) [inline Quests/Kanto/CascadeBadgeQuest.lua:47]: Adding tabulation at the end of the line? :D

File: Quests/Kanto/CascadeBadgeQuest.lua [hunk 5]
```diff
@@ -26,28 +30,39 @@ function CascadeBadgeQuest:new()
...
+function CascadeBadgeQuest:CeruleanCity()	
 	if self:needPokecenter() or not game.isTeamFullyHealed() or self.registeredPokecenter ~= "Pokecenter Cerulean" then
 		return moveToMap("Pokecenter Cerulean")
 	elseif not self:isTrainingOver() then
 		return moveToCell(39,0)-- Route 24 Bridge
 	else
-		return moveToMap("Cerulean Gym")
+		if not hasItem("Cascade Badge") then
+			return moveToMap("Cerulean Gym")
+		else  -- Gym complete --> Get Ticket Bill
```
Comment (g0ldPRO) [inline Quests/Kanto/CascadeBadgeQuest.lua:55]: This should be done before the arena.  
We have to ensure it is impossible for the bot to finish the quest without the ticket. Any time this quest start, you should check if Bill already gave us the ticket.  
That way a crash won't skip the ticket.

**Comment by Rympex** [comment]
I'm testing the new edits, i made 1 quest for do cerulean gym, ticket, and lance (vermilion city 2) and it stop on SSAnne 1F
is ok?

**Comment by g0ldPRO** [comment]
You need:
1. A quest that first gets the ticket, _then_ gets the Cerulean Badge, then enter route 5  
2. A quest that starts from route 5, talk to the guy that starts the useless vision  
3. A quest that plays the vision with only the `hasMap()` condition  
4. Continue the 2. until SSAnne  
5. A quest that does the SSAnne with only the `hasMap()` condition  
6. Continue the 2 until you have the badge, then go to Route 7 (or whatever is the name of the one to the right)  
7. A quest that gets the Flash HM (and start from the Surge arena)

Conditions:
1. `if self:hasMap() and not hasItem('Thunder Badge')` Because we will never go back there before getting the Thunder Badge. That way we can ensure the Quest will go to Route 5 even so the server crashed after getting the badge.
2. `if self:hasMap() and not hasItem('Rainbow Badge')`
3. `if self:hasMap()`
4. 2. Condition is valid as soon as we exit the vision
5. `if self:hasMap()`
6. 2. Condition is valid as soon as we exit the vision
7. `if self:hasMap() and not hasItem('Rainbow Badge')`

**Comment by Silv3rPRO** [comment]
:+1: working for me.
Selling the nugget is really cool. :smile:

**Comment by Rympex** [comment]
Edited i will post the new quest :)

**Pull request closed without merging**